.github/PULL_REQUEST_TEMPLATE/development.md

```diff
@@ -0,0 +1,33 @@
+## Pre-Review Checklist for the PR Author
+
+1. [ ] Branch follows the naming format (`iox-#123-this-is-a-branch`)
+1. [ ] Commits messages are according to this [guideline][commit-guidelines]
+    - [ ] Commit messages have the issue ID (`iox-#123 commit text`)
+    - [ ] Commit messages are signed (`git commit -s`)
+    - [ ] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
+2. [ ] Update the PR title
+   - Follow the same conventions as for commit messages
+   - Link to the relevant issue
+3. [ ] Add sensible notes for the reviewer
+4. [ ] Assign PR to reviewer
```

Comment:
Here I would add the following points:

[ ] Code formatted with clang-format.
[ ] Code according to our coding style and naming conventions.
[ ] Code change verified by unit test.
[ ] Public Code changes are documented via doxygen

Reply:
I like the idea, but maybe we can find a wording which makes it obvious that they can be ticked it they don't apply, e.g. for a PR which only fixes typos

Reply:
Rephrase the third point to:
[ ] Code behavior change verified by unit test.

So typos aren't a problem anymore.

Reply:
I'm not sure if one can rely verify that all lines have seen the clang-format. The box would be ticked if the reviewer had the feeling that it looks good? I feel that here more could be automatically checked

Reply:
The test for the changed behavior is a good thing

Reply:
> Rephrase the third point to:
> [ ] Code behavior change verified by unit test.
> 
> So typos aren't a problem anymore.

How about `[ ] Code changes verified by unit tests where applicable`. Same about the doxygen point.

Reply:
on the other side, item which are not applicable could be crossed out. We should maybe state that at the top of the template

Reply:
The third point is not clear to me. "Code change verified by unit test.".

Does this mean:
1. Unit tests are passing
2. Unit tests have been written for new behaviour ?

If it means the latter, I would change this point explicitly to "Unit tests have been written for new behaviour".

Reply:
[ ] Code formatted with clang-format.

This should not be a review point.
We need to update the workflows to check it and it should fail when clang format has not been applied.
There must be a way to do this.

Reply:
[ ] Public Code changes are documented via doxygen

This should be renamed to "Public APIs" ?
All code changes are public on github ;).

Reply:
Maybe add "[ ] Relevant issues are linked" ?

Reply:
> The third point is not clear to me. "Code change verified by unit test." (and also " Code changes verified by unit tests where applicable").
> 
> Does this mean:
> 
>     1. Unit tests are passing
> 
>     2. Unit tests have been written for new behaviour ?
> 
> 
> If it means the latter, I would change this point explicitly to "Unit tests have been written for new behaviour".

@ithier if they are not passing, the PR cannot be merged ;)